packages/bff/src/services/catalogService.ts

```diff
@@ -119,8 +173,108 @@ export function catalogServiceBuilder(
           totalCount: eservicesResponse.totalCount,
         },
       };
-
       return response;
     },
+    getProducerEServiceDescriptor: async (
+      eServiceId: string,
+      descriptorId: string,
+      context: WithLogger<BffAppContext>
+    ): Promise<bffApi.ProducerEServiceDescriptor> => {
+      const requesterId = context.authData.organizationId;
+      const headers = context.headers;
+
+      const eservice: catalogApi.EService =
+        await catalogProcessClient.getEServiceById({
+          params: {
+            eServiceId,
+          },
+          headers,
+        });
+
+      if (eservice.producerId !== requesterId) {
+        throw invalidEServiceRequester(
+          unsafeBrandId<EServiceId>(eServiceId),
+          unsafeBrandId<TenantId>(requesterId)
+        );
+      }
+
+      const descriptor = eservice.descriptors.find(
+        (e) => e.id === descriptorId
+      );
+
+      if (!descriptor) {
+        throw eserviceDescriptorNotFound(
+          unsafeBrandId<EServiceId>(eServiceId),
+          unsafeBrandId<DescriptorId>(descriptorId)
+        );
+      }
+
+      const descriptorAttributeIds: string[] = [
+        ...descriptor.attributes.certified.flatMap((atts) =>
+          atts.map((att) => att.id)
+        ),
+        ...descriptor.attributes.declared.flatMap((atts) =>
+          atts.map((att) => att.id)
+        ),
+        ...descriptor.attributes.verified.flatMap((atts) =>
+          atts.map((att) => att.id)
+        ),
+      ];
```

Comment:
Since we will reuse these checks in multiple routes I would create a `isProducer` in `validators` and `retrieveDescriptor` and `extractIdsFromAttributes` inside the service

Reply:
the others help utils methods are creted in this comment [107daaa6b9ea064d36f39e2c10069853e048af56](https://github.com/pagopa/interop-be-monorepo/pull/712/commits/107daaa6b9ea064d36f39e2c10069853e048af56)